GitCommands/Git/GitModule.cs

```diff
@@ -970,7 +970,7 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
                 objectId
             };
 
-            var revInfo = _gitExecutable.GetOutput(args, cache: GitCommandCache, outputEncoding: LosslessEncoding);
+            var revInfo = _gitExecutable.GetOutput(args, cache: objectId is null ? null : GitCommandCache, outputEncoding: LosslessEncoding);
```

Comment:
A comment here would be helpful for any future references.

Reply:
comment and variable added